pkg/controller/direct/references/resourceid.go

```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package references
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func GetResourceID(u *unstructured.Unstructured) (string, error) {
+	resourceID, _, err := unstructured.NestedString(u.Object, "spec", "resourceID")
```

Comment:
I recalled to see a team discussion about the camel case  for `projectId` rather than `projectID`. Shall this be `resourceId` or `resourceID`?
(Not a blocker to this PR, just want to align the names here since this is a shareable function under direct/references)

Reply:
So ...  go says "ID".  KRM style guide says "ID".  Google APIs use "Id" (in JSON).  Most KCC fields use `Id`, but I guess here we use `resourceID`.

If the KCC field was called `resourceId`, I don't know whether I would follow go conventions or try to match the name of the field.

In this case though, we're reading a field, and I think that field's name is `resourceID`.  That's correct against KRM style guide, but possibly inconsistent with most other GCP-generated fields.

I think it's your decision @yuwenma , and it's going to be interesting!

(Our best bet might be to stop adding fields that end in Id at all, maybe it's another marker that something is actually a ref ... or we could just drop ID arguing that it is the "type" of the field, and we don't say "fooString" so why do we say "fooID" or "fooURL" ? )